File: AD-HOC/2469/main.rb
def most_frequent
  frequency = {}
  gets.split.map(&:to_i).each do |grade|
    frequency[grade] = 0 if !frequency.include?(grade)
    frequency[grade] += 1
  end
  frequency.max_by { |grade, frequency| [frequency, grade] }[0]
end

gets
puts most_frequent

File: AD-HOC/2469/main.py
def most_frequent():
    frequency = {}
    for grade in map(int, input().split()):
        if grade not in frequency:
            frequency[grade] = 0
        frequency[grade] += 1
    return sorted(frequency.items(), key=lambda grade: [grade[1], grade[0]], reverse=True)[0][0]


input()
print(most_frequent())
